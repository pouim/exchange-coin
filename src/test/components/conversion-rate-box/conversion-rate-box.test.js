import { shallow } from 'enzyme';
import ConversionRateBox from 'src/components/ui/conversion-rate-box';

describe('ConversionRateBox', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<ConversionRateBox debug />);

        expect(component).toMatchSnapshot();
    });

    it('should render data correctly', () => {
        const testData = {
            fromCoin: {
                id: 'USD',
                title: 'USD',
                balance: 200,
                sign: '$',
            },
            toCoin: {
                id: 'EUR',
                title: 'EUR',
                balance: 150,
                sign: 'â‚¬',
            },
            conversionRate: 1.17,
        };

        const component = shallow(
            <ConversionRateBox
                fromCoin={testData.fromCoin}
                toCoin={testData.toCoin}
                conversionRate={testData.conversionRate}
            />,
        );
        const rateBoxContainer = component.find('[data-test="rate-box-container"]');

        expect(rateBoxContainer.text()).toContain(testData.fromCoin.sign);
        expect(rateBoxContainer.text()).toContain(testData.toCoin.sign);
        expect(rateBoxContainer.text()).toContain(testData.conversionRate);
    });
});
