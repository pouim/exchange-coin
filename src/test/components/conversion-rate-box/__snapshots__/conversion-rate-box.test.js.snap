// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversionRateBox should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
